sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -587,4 +587,99 @@ public static <T> T getResultWithTimeout(Future<T> future, Duration timeout)
     private static boolean hasTimeout(Duration timeout) {
         return timeout != null && !timeout.isZero() && !timeout.isNegative();
     }
+
+    /**
+     * Converts a {@link Duration} to a string in ISO-8601 format with support for a day component.
```

Comment:
It would be great if we can elaborate what the day component here is when it would be applied. Also, can we add a few unit tests? Looking at the test would give an idea of what strings will be created for different duration.